### PR TITLE
Basic Upsell System Setup + Hosting Upsell

### LINKED ISSUE #42706: Migrate to cloud Upsell
- upsell directory setup
- initial upsell components + wrappers
- first upsell: hosting

### PR DESCRIPTION
**Pull request by iethree**
Closes https://github.com/metabase/metabase/issues/42706

### Description

This establishes a new set of upsell components that are meant to be a cohesive "system" such that they are self contained and should have a consistent behavior throughout the project.

This pull request

1. Sets up the "system" in its own directory, with helpful utilities, wrappers, and documentation to enforce consistent usage.
2. Uses the system for the first instantiation: upselling cloud hosting on the setup and email settings admin pages.

Screen Shot 2024-05-16 at 9 54 44 AM
Screen Shot 2024-05-16 at 9 54 39 AM
Screen Shot 2024-05-16 at 11 03 13 AM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/admin/upsells/readme.md
Comment (iethree) [inline frontend/src/metabase/admin/upsells/readme.md:1]: @dpsutton how about this for API documentation?

File: frontend/src/metabase/admin/upsells/UpsellHosting.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    <UpsellCard
+      title={t`Get automatic updates`}
+      campaign="hosting"
+      buttonText={t`Learn more`}
+      buttonLink="https://www.metabase.com/cloud"
+      illustrationSrc={RocketGlobeIllustrationSrc}
+      source={source}
+    >
+      {jt`${(
+        <strong>{t`Migrate to Metabase cloud`}</strong>
```
Comment (rafpaf) [inline frontend/src/metabase/admin/upsells/UpsellHosting.tsx:49]:
```suggestion
        <strong>{t`Migrate to Metabase Cloud`}</strong>
```

Comment (iethree) [reply frontend/src/metabase/admin/upsells/UpsellHosting.tsx:49]: @luizarakaki ?

Comment (iethree) [reply frontend/src/metabase/admin/upsells/UpsellHosting.tsx:49]: (I had a similar thought, but all the designs had that casing)

File: frontend/src/metabase/admin/settings/components/Email/SettingsEmailForm.tsx [hunk 2]
```diff
@@ -44,15 +46,20 @@ export function SettingsEmailForm({
...
+        <SettingsSection
+          settingElements={settingElements}
+          settingValues={settingValues}
+          derivedSettingValues={derivedSettingValues}
+          updateSetting={updateSetting}
+          reloadSettings={reloadSettings}
+        />
+      </Box>
+      <Box>
+        <UpsellHosting source="settings-email-migrate_to_cloud" />
```
Comment (rafpaf) [inline frontend/src/metabase/admin/settings/components/Email/SettingsEmailForm.tsx:61]: deliberate mixing of - and _?

Comment (iethree) [reply frontend/src/metabase/admin/settings/components/Email/SettingsEmailForm.tsx:61]: yep!

File: frontend/src/metabase/admin/upsells/UpsellHosting.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    <UpsellCard
+      title={t`Minimize maintenance`}
+      campaign="hosting"
+      buttonText={t`Learn more`}
+      buttonLink="https://www.metabase.com/cloud"
+      illustrationSrc={RocketGlobeIllustrationSrc}
+      source={source}
+    >
+      {jt`${(
+        <strong>{t`Migrate to Metabase cloud`}</strong>
```
Comment (rafpaf) [inline frontend/src/metabase/admin/upsells/UpsellHosting.tsx:26]:
```suggestion
        <strong>{t`Migrate to Metabase Cloud`}</strong>
```

File: frontend/src/metabase/admin/upsells/UpsellHosting.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  if (isHosted) {
+    return null;
+  }
+
+  return (
+    <UpsellCard
+      title={t`Get automatic updates`}
+      campaign="hosting"
+      buttonText={t`Learn more`}
+      buttonLink="https://www.metabase.com/cloud"
```
Comment (rafpaf) [inline frontend/src/metabase/admin/upsells/UpsellHosting.tsx:44]: fwiw, on my local the link went to:

https://www.metabase.com/cloud/?utm_source=product&utm_medium=upsell&utm_campaign=hosting&utm_content=undefined&utm_source_plan=pro-self-hosted

Note the "undefined"

**Review by rafpaf — APPROVED** [review]
Looks great!

Note the potential issue with "undefined" appearing in the linked URL, mentioned below

File: frontend/src/metabase/admin/upsells/UpsellHosting.tsx (hunk 4 above)
Comment (iethree) [reply frontend/src/metabase/admin/upsells/UpsellHosting.tsx:44]: ah good catch, the one on the updates page was in JS so it didn't save me from myself!

**Pull request merged**